src/utilities/use-toggle.ts

```diff
@@ -0,0 +1,24 @@
+import {useState, useCallback} from 'react';
+
+export function useToggle(initialState: boolean) {
```

Comment:
I can see why you did this, I just wonder it's worth it. It's simple but it requires knowledge about the hook and it's less declarative and just handling the toggle when needed. We aren't saving on much here imo.

Reply:
The same can be said for pretty much all of our custom hooks :)

IMO The value from these small utilities comes from encapsulation, that small amount of complexity saved adds up over the course of several components and its repeated presence means we'll become familiar with it over time for when we have a boolean toggle, which is pretty often.

Reply:
The same can be said for pretty much all of our custom hooks :)

IMO The value from these small utilities comes from encapsulation, that small amount of complexity saved adds up over the course of several components and its repeated presence means we'll become familiar with it over time for when we have a boolean toggle, which is pretty often.

Reply:
I think for a really common usecase like a toggle variable it's fine. I agree that every hook adds to the cognitive load needed to understand the codebase, and we should avoid them